domain/application/service/charm.go

```diff
@@ -886,6 +886,17 @@ func (s *Service) setCharm(ctx context.Context, args charm.SetCharmArgs) (setCha
 	}, warnings, nil
 }
 
+// ReserveCharmRevision reserves a charm revision for the given charm id.
```

Comment:
Not great wording. There's no _given_ charm ID.

I'd say something more along the lines of creating a placeholder charm entry based on the input `SetCharmArgs` and returning the ID.